src/analysis/jump.ml

```diff
@@ -127,19 +127,54 @@ let rec skip_non_moving pos = function
   | [] -> []
 ;;
 
+let rec find_case_pos pos cases cmp_fn =
+  match cases with
+  | [] -> None
+  | { c_lhs = { pat_loc; _ }; _ } :: tail ->
+    let case_loc = pat_loc.loc_start in
+    if cmp_fn pos case_loc then
+      Some case_loc
+    else
+      find_case_pos pos tail cmp_fn
+
+let get_match_case_pos node pos cmp_fn match_target =
+  try
+    match match_pred [node] with
+    | Some (Expression { exp_desc = Texp_match (_, cases, _); _ }) ->
+      let case = find_case_pos pos cases cmp_fn in      
+      (match case with
+      | Some location ->
+        `Found location
+      | None -> (match match_target with 
+            | "match-next-case" -> `Error "No next case found"
+            | "match-prev-case" -> `Error "No previous case found"
+            | _ -> raise (No_matching_target)))
```

Comment:
This last case is not reachable. This highlight the fact that passing [match_target] to that function is not a good fit. Instead you can make a new type `type direction = Prev | Next` and pass this more precise information.